src/api/app/models/review.rb

```diff
@@ -19,7 +19,14 @@ class NotFoundError < APIError
   validates :reviewer, length: { maximum: 250 }
   validates :reason, length: { maximum: 65_534 }
 
-  validate :check_initial, on: [:create]
+  validates :user, presence: true, if: :by_user?
+  validates :group, presence: true, if: :by_group?
+  validates :project, presence: true, if: :by_project?
+  validates :package, presence: true, if: :by_package?
+
+  validate :review_assignment
+  validate :correctness_of_package, if: :by_package?
```

Comment:
project is always required now, this is not needed (it won't be executed if project is not there anyway)

Reply:
The difference is that this validation checks that `by_project` is set when `by_package` is not nil. The other validations are checking the object relations, and only on review creation.

But we can write `correctness_of_package` with the default rails validations.... *updated*